pkg/bgpv1/agent/state.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package agent
+
+// SessionState as defined in rfc4271#section-8.2.2
+type SessionState uint32
+
+const (
+	SessionUnknown SessionState = iota
+	SessionIdle
+	SessionConnect
+	SessionActive
+	SessionOpenSent
+	SessionOpenConfirm
+	SessionEstablished
+)
+
+func (s SessionState) String() string {
+	switch s {
+	case SessionUnknown:
+		return "unknown"
+	case SessionIdle:
+		return "idle"
+	case SessionConnect:
+		return "connect"
+	case SessionActive:
+		return "active"
+	case SessionOpenSent:
+		return "open sent"
+	case SessionOpenConfirm:
+		return "open confirm"
```

Comment:
Nit: Let's use underscore as a word delimiter for the consistency with address family.